src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -254,5 +210,139 @@ public void WriteTo(TextWriter writer, HtmlEncoder encoder)
         }
 
         private string DebuggerToString() => _name;
+
+        public void CopyTo(IHtmlContentBuilder destination)
+        {
+            if (Pages == null)
+            {
+                return;
+            }
+
+            for (var i = 0; i < Pages.Count; i++)
+            {
+                var page = Pages[i];
+                for (var j = 0; j < page.Count; j++)
+                {
+                    var value = page.Buffer[j];
+
+                    string valueAsString;
+                    IHtmlContentContainer valueAsContainer;
+                    if ((valueAsString = value.Value as string) != null)
+                    {
+                        destination.AppendHtml(valueAsString);
+                    }
+                    else if ((valueAsContainer = value.Value as IHtmlContentContainer) != null)
+                    {
+                        valueAsContainer.CopyTo(destination);
+                    }
+                    else
+                    {
+                        destination.AppendHtml((IHtmlContent)value.Value);
+                    }
+                }
+            }
+        }
+
+        public void MoveTo(IHtmlContentBuilder destination)
+        {
+            if (Pages == null)
+            {
+                return;
+            }
+
+            var other = destination as ViewBuffer;
+            if (other != null)
+            {
+                MoveTo(other);
+                return;
+            }
+
+            for (var i = 0; i < Pages.Count; i++)
+            {
+                var page = Pages[i];
+                for (var j = 0; j < page.Count; j++)
+                {
+                    var value = page.Buffer[j];
+
+                    string valueAsString;
+                    IHtmlContentContainer valueAsContainer;
+                    if ((valueAsString = value.Value as string) != null)
+                    {
+                        destination.AppendHtml(valueAsString);
+                    }
+                    else if ((valueAsContainer = value.Value as IHtmlContentContainer) != null)
+                    {
+                        valueAsContainer.MoveTo(destination);
+                    }
+                    else
+                    {
+                        destination.AppendHtml((IHtmlContent)value.Value);
+                    }
+                }
+            }
+
+            for (var i = 0; i < Pages.Count; i++)
+            {
+                var page = Pages[i];
+                Array.Clear(page.Buffer, 0, page.Capacity);
```

Comment:
It can contain arbitrary objects. If this page is returned to the pool, they could hang out for an arbitrary amount of time. Basically I'm being extra careful